File: apps/patter-voice-sdk/libraries/python/getpatter/stt/cartesia.py
"""Cartesia streaming STT for Patter pipeline mode."""

from __future__ import annotations

import os
from typing import ClassVar

from getpatter.providers.cartesia_stt import CartesiaSTT as _CartesiaSTT

__all__ = ["STT"]


class STT(_CartesiaSTT):
    """Cartesia ``ink-whisper`` streaming STT.

    Example::

        from getpatter.stt import cartesia

        stt = cartesia.STT()                # reads CARTESIA_API_KEY
        stt = cartesia.STT(api_key="...", language="es")
    """

    provider_key: ClassVar[str] = "cartesia_stt"

    def __init__(
        self,
        api_key: str | None = None,
        *,
        language: str = "en",
        model: str = "ink-whisper",
        sample_rate: int = 16000,
    ) -> None:
        key = api_key or os.environ.get("CARTESIA_API_KEY")
        if not key:
            raise ValueError(
                "Cartesia STT requires an api_key. Pass api_key='...' or "
                "set CARTESIA_API_KEY in the environment."
            )
        super().__init__(
            api_key=key,
            language=language,
            model=model,
            sample_rate=sample_rate,
        )

File: apps/patter-voice-sdk/libraries/typescript/src/stt/cartesia.ts
/** Cartesia streaming STT for Patter pipeline mode. */
import { CartesiaSTT as _CartesiaSTT, type CartesiaEncoding } from "../providers/cartesia-stt";

/** Constructor options for the Cartesia `STT` adapter. */
export interface CartesiaSTTOptions {
  /** API key. Falls back to CARTESIA_API_KEY env var when omitted. */
  apiKey?: string;
  model?: string;
  language?: string;
  encoding?: CartesiaEncoding;
  sampleRate?: number;
  baseUrl?: string;
}

/**
 * Cartesia streaming STT (ink-whisper).
 *
 * @example
 * ```ts
 * import * as cartesia from "getpatter/stt/cartesia";
 * const stt = new cartesia.STT();              // reads CARTESIA_API_KEY
 * const stt = new cartesia.STT({ apiKey: "..." });
 * ```
 */
export class STT extends _CartesiaSTT {
  static readonly providerKey = "cartesia_stt";
  constructor(opts: CartesiaSTTOptions = {}) {
    const key = opts.apiKey ?? process.env.CARTESIA_API_KEY;
    if (!key) {
      throw new Error(
        "Cartesia STT requires an apiKey. Pass { apiKey: '...' } or " +
          "set CARTESIA_API_KEY in the environment.",
      );
    }
    super(key, {
      model: opts.model,
      language: opts.language,
      encoding: opts.encoding,
      sampleRate: opts.sampleRate,
      baseUrl: opts.baseUrl,
    });
  }
}
